bin/activate.bat

```diff
@@ -1,68 +1,54 @@
 @echo off
+setlocal
 
-for /f "delims=" %%i in ("%~dp0..\envs") do (
-    set ANACONDA_ENVS=%%~fi
-)
-
-set CONDA_NEW_NAME=%~1
+set CONDA_NEW_ENV=%1
+set CONDA_NEW_ENV=%CONDA_NEW_ENV:"=%
 
-if "%~2" == "" goto skiptoomanyargs
-    echo ERROR: Too many arguments provided
-    goto usage
+if "%2" == "" goto skiptoomanyargs
+    (echo Error: did not expect more than one argument.) 1>&2
+    exit /b 1
 :skiptoomanyargs
 
-if "%CONDA_NEW_NAME%" == "" set CONDA_NEW_NAME=%~dp0..\
-
-if exist "%ANACONDA_ENVS%\%CONDA_NEW_NAME%\conda-meta" goto usenamedenv
-    for /F %%i in ("%CONDA_NEW_NAME%") do set CONDA_NEW_PATH=%%~fi
-    if exist "%CONDA_NEW_PATH%\conda-meta" goto usefullpath
-        echo No environment named "%CONDA_NEW_NAME%" exists in %ANACONDA_ENVS%
-        set CONDA_NEW_NAME=
-        set CONDA_NEW_PATH=
-        exit /b 1
-:usenamedenv
-    set CONDA_NEW_PATH=%ANACONDA_ENVS%\%CONDA_NEW_NAME%
-    goto skipmissingenv
-:usefullpath
-    for /F %%i in ("%CONDA_NEW_PATH%") do set CONDA_NEW_NAME=%%~ni
-:skipmissingenv
+if not "%1" == "" goto skipmissingarg
+    (echo Error: no environment provided.) 1>&2
+    exit /b 1
+:skipmissingarg
+
+if not "%1" == "--help" goto skipusage
+    (
+    echo Usage: activate ENV
+    echo.
+    echo Deactivates previously activated Conda
+    echo environment, then activates the chosen one.
+    ) 1>&2
+    exit /b 1
+:skipusage
+
+REM Use conda itself to figure things out
+REM No conda, no dice
+
+SET CONDAFOUND=
+for %%X in (conda.exe) do (set CONDAFOUND=%%~$PATH:X)
+if not defined CONDAFOUND for %%X in (conda.bat) do (set CONDAFOUND=%%~$PATH:X)
+if defined CONDAFOUND goto runcondasecretcommand
+    echo "cannot find conda to test for the environment %CONDA_NEW_ENV%"
+    exit /b 1
+
+:runcondasecretcommand
+REM Run secret conda ..checkenv command
+call "%CONDAFOUND%" ..checkenv %CONDA_NEW_ENV%
+REM EQU 0 means 0 or above on Windows ;(
+if %ERRORLEVEL% EQU 1 (
+    exit /b 1
+)
 
 REM Deactivate a previous activation if it is live
-if "%CONDA_DEFAULT_ENV%" == "" goto skipdeactivate
-    REM This search/replace removes the previous env from the path
-    echo Deactivating environment "%CONDA_DEFAULT_ENV%"...
-
-    REM Run any deactivate scripts
-    if not exist "%CONDA_DEFAULT_ENV%\etc\conda\deactivate.d" goto nodeactivate
-        pushd "%CONDA_DEFAULT_ENV%\etc\conda\deactivate.d"
-        for %%g in (*.bat) do call "%%g"
-        popd
-    :nodeactivate
+FOR /F "delims=" %%i IN ('"%CONDAFOUND%" ..deactivate') DO set PATH=%%i
 
-    set CONDACTIVATE_PATH="%CONDA_DEFAULT_ENV%";"%CONDA_DEFAULT_ENV%\Scripts";"%CONDA_DEFAULT_ENV%\Library\bin"
```

Comment:
I don't see that you are keeping the \Scripts and \Library\bin folders on PATH with your new scheme.  These are pretty important to have.  Can you add them, or explain how your script keeps them?